x-pack/plugins/transform/public/app/sections/transform_management/components/transform_list/transform_list.tsx

```diff
@@ -114,90 +103,41 @@ export const TransformList: FC<Props> = ({
     !capabilities.canPreviewTransform ||
     !capabilities.canStartStopTransform;
 
-  const onQueryChange = ({
-    query,
-    error,
-  }: Parameters<NonNullable<EuiSearchBarProps['onChange']>>[0]) => {
-    if (error) {
-      setSearchError(error.message);
+  const { columns, modals: singleActionModals } = useColumns(
+    expandedRowItemIds,
+    setExpandedRowItemIds,
+    transformSelection
+  );
+
+  const setQueryClauses = (queryClauses: any) => {
+    if (queryClauses.length) {
+      const filtered = filterTransforms(transforms, queryClauses);
+      setFilteredTransforms({ active: true, items: filtered });
     } else {
-      let clauses: Clause[] = [];
-      if (query && query.ast !== undefined && query.ast.clauses !== undefined) {
-        clauses = query.ast.clauses;
-      }
-      if (clauses.length > 0) {
-        setFilterActive(true);
-        filterTransforms(clauses as Array<TermClause | FieldClause>);
-      } else {
-        setFilterActive(false);
-      }
-      setSearchError(undefined);
+      setFilteredTransforms({ active: false, items: [] });
     }
   };
 
-  const filterTransforms = (clauses: Array<TermClause | FieldClause>) => {
-    setIsLoading(true);
-    // keep count of the number of matches we make as we're looping over the clauses
-    // we only want to return transforms which match all clauses, i.e. each search term is ANDed
-    // { transform-one:  { transform: { id: transform-one, config: {}, state: {}, ... }, count: 0 }, transform-two: {...} }
-    const matches: Record<string, any> = transforms.reduce((p: Record<string, any>, c) => {
-      p[c.id] = {
-        transform: c,
-        count: 0,
-      };
-      return p;
-    }, {});
-
-    clauses.forEach((c) => {
-      // the search term could be negated with a minus, e.g. -bananas
-      const bool = c.match === 'must';
-      let ts = [];
-
-      if (c.type === 'term') {
-        // filter term based clauses, e.g. bananas
-        // match on ID and description
-        // if the term has been negated, AND the matches
-        if (bool === true) {
-          ts = transforms.filter(
-            (transform) =>
-              stringMatch(transform.id, c.value) === bool ||
-              stringMatch(transform.config.description, c.value) === bool
-          );
-        } else {
-          ts = transforms.filter(
-            (transform) =>
-              stringMatch(transform.id, c.value) === bool &&
-              stringMatch(transform.config.description, c.value) === bool
-          );
+  useEffect(() => {
+    const filterList = () => {
+      if (searchQueryText !== '') {
+        const query = EuiSearchBar.Query.parse(searchQueryText);
+        let clauses: any = [];
```

Comment:
Suggestion to use Array<TermClause | FieldClause> here if that's what `filterAnalytics` accepts

Reply:
Good call - I think this also might be part of the EUI type issue I ran into but I will timebox some further digging time 👌